PocketKit/Sources/PocketKit/UserManagementService.swift

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SharedPocketKit
+import Sync
+import Combine
+
+protocol UserManagementServiceProtocol {
+    func deleteAccount() async throws
+    func logout()
+}
+
+struct UserManagementService: UserManagementServiceProtocol {
+    let appSession: AppSession
+    let user: User
+    let notificationCenter: NotificationCenter
+    let source: Source
+
+    func deleteAccount() async throws {
+        try await source.deleteAccount()
+        await logout()
+        self.notificationCenter.post(name: .userDeleted, object: nil)
```

Comment:
I wonder if it would be possible to use a Combine `Publisher` instead of posting a notification? Would this service be injected into any object that needs it?